File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminTimePreference.java
package org.digitalcampus.oppia.utils.custom_prefs;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.MultiSelectListPreference;

public class AdminTimePreference extends TimePreference implements AdminPreference {

    private OnPreferenceClickListener customOnPreferenceClickListener;

    public AdminTimePreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onClick() {

        if (customOnPreferenceClickListener != null) {
            boolean handled = customOnPreferenceClickListener.onPreferenceClick(this);
            if (handled) {
                return;
            }
        }

        super.onClick();

    }

    @Override
    public void setOnPreferenceClickListener(OnPreferenceClickListener onPreferenceClickListener) {
        customOnPreferenceClickListener = onPreferenceClickListener;
    }

    @Override
    public void onAccessGranted() {
        super.onClick();
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminTimePreference.kt
package org.digitalcampus.oppia.utils.custom_prefs

import android.content.Context
import android.util.AttributeSet

class AdminTimePreference(context: Context, attrs: AttributeSet) : TimePreference(context, attrs), AdminPreference {
    private var customOnPreferenceClickListener: OnPreferenceClickListener? = null
    override fun onClick() {
        if (customOnPreferenceClickListener != null) {
            val handled = customOnPreferenceClickListener!!.onPreferenceClick(this)
            if (handled) {
                return
            }
        }
        super.onClick()
    }

    override fun setOnPreferenceClickListener(onPreferenceClickListener: OnPreferenceClickListener?) {
        customOnPreferenceClickListener = onPreferenceClickListener
    }

    override fun onAccessGranted() {
        super.onClick()
    }
}